### PR TITLE
fix: better timers management for race conditions and leak prevention

### PR DESCRIPTION
**Pull request by robertsLando**

### CODE REVIEW

**Comment by robertsLando** [comment]
Wait before merging, I still have some things to check. Give it a look in the meanwhile to tell me your opinion on this @Supergiovane

**Comment by Supergiovane** [comment]
Hi Daniel
it seems to be ok, but only the time will tell us.

File: src/KNXClient.ts [hunk 1]
```diff
@@ -349,6 +358,51 @@ export default class KNXClient extends TypedEventEmitter<KNXClientEventCallbacks
...
 
+	/** Waits till providden event occurs for at most the providden timeout */
+	private async waitForEvent(event: KNXClientEvents, timeout: number) {
+		return Promise.race<void>([
+			new Promise((resolve) => {
+				this.once(event, resolve)
+			}),
+			wait(timeout),
+		])
+	}
```
[the remark this answers is not in the document]
Comment (robertsLando) [reply src/KNXClient.ts:369]: Not yet, maybe when I will start looking at secure things :)

**Comment by robertsLando** [comment]
Once this is merged I will try to make a new release my own to see if I have required permissions

**Comment by robertsLando** [comment]
@Supergiovane sure!

**Comment by robertsLando** [comment]
Release has gone well! Perfect :)

**Pull request merged**